azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTestBase.java

```diff
@@ -170,4 +171,8 @@ protected void succeedJobs(final String... jobs) {
     }
   }
 
+  protected void assertEvents(final Type... types) {
```

Comment:
Yes, we can agree to disagree. I am hoping to learn from this discussion and your perspective too. It doesn't mean that we have to agree on everything.

With the help of IDEs, I find it easier nowadays to optimize for reading than writing. 

The code duplication you referenced is much less than before. Yes, I'd like to further simplify the code. 
"this.flowMap = FlowRunnerTestUtil
    .prepareProject(this.project, dir, this.logger, this.workingDir);
"
Again, I would prefer an incremental approach, especially when dealing with legacy code which I don't understand well enough.  I would welcome any suggestion on how to simplify this logic further. 

Regarding the overhead, I tried the approach myself see 
#1147 with the help of intellij and your quick review, it didn't take much longer to make two PRs than one.
I understand the overhead. Again I'd like to optimize for reading than writing. After all, we read the code more often than modifying it. 

Thanks again for your help and understanding!